x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/predicate/operator/comparison/LessThanOrEqual.java

```diff
@@ -8,8 +8,54 @@
 
 import org.apache.lucene.util.BytesRef;
 import org.elasticsearch.compute.ann.Evaluator;
+import org.elasticsearch.xpack.ql.expression.Expression;
+import org.elasticsearch.xpack.ql.expression.FieldAttribute;
+import org.elasticsearch.xpack.ql.expression.TypeResolutions;
+import org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.BinaryComparison;
+import org.elasticsearch.xpack.ql.tree.NodeInfo;
+import org.elasticsearch.xpack.ql.tree.Source;
+import org.elasticsearch.xpack.ql.type.DataTypes;
+
+import java.time.ZoneId;
+
+public class LessThanOrEqual extends org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.LessThanOrEqual {
+    public LessThanOrEqual(Source source, Expression left, Expression right, ZoneId zoneId) {
+        super(source, left, right, zoneId);
+    }
+
+    @Override
+    protected TypeResolution resolveInputType(Expression e, TypeResolutions.ParamOrdinal paramOrdinal) {
+        if (e instanceof FieldAttribute fa && fa.dataType() == DataTypes.TEXT) {
+            return TypeResolution.TYPE_RESOLVED;
+        }
+        return super.resolveInputType(e, paramOrdinal);
+    }
+
+    @Override
+    protected NodeInfo<org.elasticsearch.xpack.ql.expression.predicate.operator.comparison.LessThanOrEqual> info() {
+        return NodeInfo.create(this, LessThanOrEqual::new, left(), right(), zoneId());
+    }
+
+    @Override
+    protected LessThanOrEqual replaceChildren(Expression newLeft, Expression newRight) {
+        return new LessThanOrEqual(source(), newLeft, newRight, zoneId());
+    }
+
+    @Override
+    public GreaterThanOrEqual swapLeftAndRight() {
+        return new GreaterThanOrEqual(source(), right(), left(), zoneId());
+    }
+
+    @Override
+    public GreaterThan negate() {
+        return new GreaterThan(source(), left(), right(), zoneId());
+    }
+
+    @Override
+    public BinaryComparison reverse() {
+        return super.reverse();
```

Comment:
This will return a QL GT.
 
I'm wondering if all these functions shouldn't rather extend QL's BinaryComparsion directly, since they override most methods anyways. Or alterantively, there could be an EsqlBinaryComparion (extending QL's), which could implement the type resolution, which is identical in all the new comparisons.

Reply:
Argh! The whole point of extending that method was to return the right instance, and I just missed it.
Fixing it right away